Vwitter/View Controllers/ProfileViewController.m

```diff
@@ -0,0 +1,75 @@
+//
+//  ProfileViewController.m
+//  Vwitter
+//
+//  Created by Christina Li on 7/8/22.
+//
+
+#import "ProfileViewController.h"
+#import <Parse/Parse.h>
+#import "Vent.h"
+#import "VentCell.h"
+
+@interface ProfileViewController () <UITableViewDataSource, UITableViewDelegate>
+@property (weak, nonatomic) IBOutlet UILabel *username;
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+@property (strong, nonatomic) NSMutableArray<Vent *> *arrayOfVents;
+
+@end
+
+@implementation ProfileViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view.
+    self.username.text = [PFUser currentUser].username;
+    
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    
+    [self loadData];
+}
+
+- (void)loadData {
+    PFQuery *vaQuery = [Vent query];
+    [vaQuery orderByDescending:@"createdAt"];
+    [vaQuery whereKey:@"author" equalTo:[PFUser currentUser]];
+    [vaQuery includeKey:@"author"];
+    
+    vaQuery.limit = 20;
+    
+    __weak typeof(self) weakSelf = self;
+    [vaQuery findObjectsInBackgroundWithBlock:^(NSArray<Vent *> * _Nullable vents, NSError * _Nullable error) {
+        typeof(self) strongSelf = weakSelf;
+        if (!strongSelf) {
+            NSLog(@"I got killed!");
+            return;
+        }
+        if (vents) {
```

Comment:
I think the `findObjectsInBackgroundWithBlock` method just returns an untyped object (`id`) right, did you change it to be `NSArray`? I would not change it to be an array, because you don't own this API and cannot guarantee that the object it returns will actually be an `NSArray`. Instead, I would use the `CAST_TO_CLASS_OR_NIL ` macro you added to be safe.

```
NSArray<Vent *> *_Nullable ventArray = CAST_TO_CLASS_OR_NIL(object, NSArray);
if (ventArray) {

...
```